iphone/cli/commands/_build.js

```diff
@@ -2990,7 +2990,8 @@ iOSBuilder.prototype.createXcodeProject = function createXcodeProject(next) {
 			DEAD_CODE_STRIPPING: 'YES',
 			SDKROOT: 'iphoneos',
 			CODE_SIGN_ENTITLEMENTS: '"' + appName + '.entitlements"'
-		};
+		},
+		legacySwift = version.lt(this.xcodeEnv.version, '8.0.0');
```

Comment:
Please note that the legacy Swift 2.3 can only be used until Xcode 8.2.1. They removed Swift 2.3 from Xcode 8.3 and later, classic Apple move. We should (very soon) only allow Swift 3.0 and later, so we are ready for Swift 4 already - which will be backwards-compatible to Swift 3.1.